db/repositories/generated_feed_repository.py

```
@@ -0,0 +1,133 @@
+            feed: GeneratedFeed model to create or update
+            
+        Returns:
+            The created or updated GeneratedFeed object
+            
+        Raises:
+            ValueError: If agent_handle, run_id, or turn_number is empty
+            sqlite3.IntegrityError: If composite key violates constraints (from adapter)
+            sqlite3.OperationalError: If database operation fails (from adapter)
+        """
+        if not feed.agent_handle or not feed.agent_handle.strip():
+            raise ValueError("agent_handle cannot be empty")
```

Ruff (0.14.6)

Avoid specifying long messages outside the exception class

(TRY003)